plugin/sycl/objective/multiclass_obj.cc

```diff
@@ -66,51 +84,62 @@ class SoftmaxMultiClassObj : public ObjFunction {
           << "Number of weights should be equal to number of data points.";
     }
 
-    ::sycl::buffer<bst_float, 1> preds_buf(preds.HostPointer(), preds.Size());
-    ::sycl::buffer<bst_float, 1> labels_buf(info.labels.Data()->HostPointer(), info.labels.Size());
-    ::sycl::buffer<GradientPair, 1> out_gpair_buf(out_gpair->Data()->HostPointer(),
-                                                  out_gpair->Size());
-    ::sycl::buffer<bst_float, 1> weights_buf(is_null_weight ? NULL : info.weights_.HostPointer(),
-                                             is_null_weight ? 1 : info.weights_.Size());
+    bst_float* preds_ptr = preds_.Data();
+    bst_float* labels_ptr = labels_.Data();
+    bst_float* weights_ptr = weights_.Data();
+    GradientPair* out_gpair_ptr = out_gpair_.Data();
+
+    int label_correctness_flag = 1;
+    const size_t wg_size = 32;
+    const size_t nBatch = ndata / kBatchSize + (ndata % kBatchSize > 0);
```

Comment:
Is it possible that we can have a separate function or a set of helpers for calculating batches that are independent of the objective functions? It would be great if we could start the objective function with a simpler structure and avoid things that split out things that are not strictly related to the math itself (like batch, memcpy, etc).

Reply:
Hi, I have introduced the `BatchProcessingHelper `class. It conceals most of the details regarding batch processing implementation.